bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelStateDescriptionProvider.java

```diff
@@ -57,7 +61,29 @@ public StateDescription getStateDescription(String itemName, Locale locale) {
             Channel channel = thingRegistry.getChannel(channelUID);
             if (channel != null) {
                 ChannelType channelType = thingTypeRegistry.getChannelType(channel, locale);
-                return channelType != null ? channelType.getState() : null;
+                if (channelType != null) {
+                    StateDescription state = channelType.getState();
+                    if ((channelType.getItemType() != null) && ((state == null) || (state.getPattern() == null))) {
+                        String pattern = null;
+                        if (channelType.getItemType().equalsIgnoreCase("String")) {
```

Comment:
Interesting. This is against our own guidelines, so feel free to add this export as part of this PR :-)